pkg/services/object/get/assembly_v2.go

```diff
@@ -116,7 +116,7 @@ func (exec *execCtx) rangeFromLink(link objectSDK.Link) bool {
 				rngPerChild.SetOffset(uint64(firstOffset))
 			}
 			if i == last {
-				rngPerChild.SetLength(uint64(child.ObjectSize()) - uint64(lastBound+firstOffset))
+				rngPerChild.SetLength(uint64(lastBound - firstOffset))
```

Comment:
Isn't it just `uint64(lastBound)`?

Reply:
we may range only part of the first object, then we do not need its offset

Reply:
But what if we have multiple objects? You get `X:$END` of `first`, OK, then you go over some set of complete objects and then you need `0:Y`, while you do `0:Y-X`. Or not?

Reply:
hm, seems like you are right but why do tests work then? we never range over the bounds in the tests?